libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -812,7 +812,7 @@ protected TokenEndpointResponse performRequest(TokenEndpointResponse tr) throws
                     HttpAccess.DEFAULT, tr.idUrlWithInstance, tr.authToken);
 
                 // Request the authenticated user's information to determine if it is a Salesforce integration user.  This is a synchronous network request, so it must be performed here in the background stage.
-                isSalesforceIntegrationUser = fetchIsSalesforceIntegrationUser(tr);
+                isSalesforceIntegrationUser = SalesforceSDKManager.getInstance().shouldBlockSalesforceIntegrationUser() && fetchIsSalesforceIntegrationUser(tr);
```

Comment:
NIT could we rename the variable since it might not always represent whether the user is indeed an integration user.
Maybe call it `shouldBlockUser` ?
Then line 677 of `OauthWebViewHelper.java`, we could simply do `if(shouldBlockUser)`

Reply:
I was on the fence of doing just that earlier also.  Thanks for the encouragement to boost it into action.  How's this? 👉🏻 https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2417/commits/5d802437ee7d965c70b8648d0a91dc60d286ebdd